### DISCUSSION TITLE
add an expression to join on

### DISCUSSION THREAD

**Discussion by zhujianwei-star** [JSQLParser/JSqlParser#1828] [General]

**Comment by zhujianwei-star** [comment]
Now, I will add an expression to join on, i should use function to get a collections of expresssions, but i can not add an expression to join on by add an expressions to the collections, because It will rely on the ON keyword and the result will be an error when executed in mysql. I wish it can rely on the AND keyword.
![image](https://github.com/JSQLParser/JSqlParser/assets/58215451/f93863c9-d17d-4b59-9ca4-5eacef558564)

**Comment by manticore-projects** [comment]
Greetings.

1) Create an [`AndExpression`](https://manticore-projects.com/JSQLParser/javadoc_snapshot.html#andexpression) from two [`EqualsTo`](https://manticore-projects.com/JSQLParser/javadoc_snapshot.html#equalsto) Expressions
2) Assign this `AndExpression` to the `Join.onExpression`

**Comment by zhujianwei-star** [comment]
I know I can solve the problem this way, the code I commented out does, but what I want is to be able to solve it directly by setting an expression in the collection, which is a lot easier and saves a lot of useless code.Otherwise, why would you convert getting getOnExpression() to the getOnExpressions() method

**Comment by zhujianwei-star** [comment]
it just a direction that can be optimized, otherwise I don't feel there's much point in getting the collection

**Comment by manticore-projects** [comment]
I am sorry. I have actually zero idea what you are trying to ask for or what exactly you want to suggest.
`OnExpressions` is a Collection for a purpose since there are statements where a list is needed:

```sql
select *
from a inner join b inner join c on a.a=b.a on b.c = c.c
```

Furthermore, you can simple use the `addOnExpression()` and `getOnExpressions().get(0)` methods to access the first element of that list.

However, if you see a potential for improving the API, please send a PR.

**Comment by zhujianwei-star** [comment]
I am sorry. I did not think anout this. Now, i have no problem, thank you for your explain. If I think of a good solution to this problem I raised myself, I am coming to consult you or send a PR.

**Comment by manticore-projects** [comment]
You are most welcome. Thank you for your interest.
